server/services/applicationService.ts

```diff
@@ -80,6 +87,37 @@ export default class ApplicationService {
     }
   }
 
+  private getResponsesForTask(application: Application, taskName: string): Array<PageResponse> {
+    const pageNames = Object.keys(application.data[taskName])
+    const responsesForPages = pageNames.map(pageName => this.getResponseForPage(application, taskName, pageName))
+    return responsesForPages
+  }
+
+  private getResponseForPage(application: Application, taskName: string, pageName: string): PageResponse {
+    const Page = this.getPage(taskName, pageName)
+
+    const body = application?.data?.[taskName]?.[pageName]
+    const page = new Page(body, application)
```

Comment:
I think having body as undefined (as it could be if the `application`, `data`,  or `[taskName]` are undefined) could cause issues further down the line (EG in the constructor of `placementDate`). Might be worth throwing a `SessionDataError` here?

Reply:
I think it might be worth looking at later when we start using this in anger 👍